### PR TITLE
IGVF-292-treatment-purpose-required

### PR DESCRIPTION
**Pull request by khzlin**

### CODE REVIEW

File: src/igvfd/upgrade/treatment.py [hunk 1]
```diff
@@ -10,3 +10,9 @@ def treatment_1_2(value, system):
     if 'documents' in value:
         if len(value['documents']) == 0:
             del value['documents']
+
+
+@upgrade_step('treatment', '2', '3')
```
Comment (gabdank) [inline src/igvfd/upgrade/treatment.py:15]: i think cleaner would be to turn any treatment not specifying purpose to be "perturbation" and add that info to "notes".

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/schemas/changelogs/treatment.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 ## Changelog for *`treatment.json`*
 
+### Schema version 3
+* Add `purpose` to requirements.
```
[the remark this answers is not in the document]
Comment (khzlin) [reply src/igvfd/schemas/changelogs/treatment.md:4]: ok

File: src/igvfd/upgrade/treatment.py [hunk 3]
```diff
@@ -10,3 +10,12 @@ def treatment_1_2(value, system):
...
+def treatment_2_3(value, system):
+    # https://igvf.atlassian.net/browse/IGVF-292
+    if 'purpose' not in value:
+        value['purpose'] = 'perturbation'
+        value['notes'] = 'This treatment did not have purpose specified previously, it was upgraded to have perturbation purpose.'
```
Comment (gabdank) [inline src/igvfd/upgrade/treatment.py:20]: @khzlin you can not assume the notes field was empty, so need to concatenate rather than overwrite it

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/upgrade/treatment.py [hunk 4]
```diff
@@ -10,3 +10,14 @@ def treatment_1_2(value, system):
...
+
+
+@upgrade_step('treatment', '2', '3')
+def treatment_2_3(value, system):
+    # https://igvf.atlassian.net/browse/IGVF-292
+    notes = value.get('notes', '')
+    if 'purpose' not in value:
+        value['purpose'] = 'perturbation'
+        value['notes'] = (
+            notes + 'This treatment did not have purpose specified previously, it was upgraded to have perturbation purpose.').strip()
```
Comment (gabdank) [inline src/igvfd/upgrade/treatment.py:22]: if 'notes' in value:
            value['notes'] = f'{value.get('notes')}. This treatment did not have..........'
else:
      value['notes'] = 'This treatment did not have..........'

**Review by gabdank — CHANGES_REQUESTED** [review]

**Review by gabdank — APPROVED** [review]

**Pull request merged**